lms/models/_display_name.py

```diff
@@ -0,0 +1,27 @@
+def display_name(given_name, family_name, full_name):
+    """
+    Return an h-compatible display name the given name parts.
+
+    LTI 1.1 launch requests have separate given_name (lis_person_name_given),
+    family_name (lis_person_name_family) and full_name (lis_person_name_full)
+    parameters. This function returns a single display name string based on
+    these three separate names.
+    """
+    name = full_name.strip()
+
+    if not name:
+        given_name = given_name.strip()
+        family_name = family_name.strip()
+
+        name = " ".join((given_name, family_name)).strip()
+
+    if not name:
+        return "Anonymous"
+
+    # The maximum length of an h display name.
+    display_name_max_length = 30
+
+    if len(name) <= display_name_max_length:
+        return name
+
+    return name[: display_name_max_length - 1].rstrip() + "…"
```

Comment:
This is `_display_name.py` because the module only contains collaborators used internally within `lms.models`, it's not meant to be used outside of `lms.models`. The function is named `display_name()` (no underscore) because it's part of the `lms.models._display_name` module's public interface: it's meant to be called by code outside of the module.